elastic-agent.yml

```diff
@@ -261,6 +261,11 @@ agent.logging.to_stderr: true
 # and conditionals. Each provider's keys are automatically prefixed with the name
 # of the provider.
 
+# All registered providers are enabled by default.
+
+# Disable all providers by default and only enable explicitly configured providers.
+# providers_default_disable: true
```

Comment:
I don't really like this as top-level field. The agent has a couple sections in its configuration `outputs`, `inputs`, `management`, `agent`. This as a top-level option is kinda floating out in the wrong area. Being that you are changing the default option of the Elastic Agent for all provides I would think this should be under `agent.*`.

What about: `agent.providers.initial_default: false`? The `initial_default` can be a `*bool` to allow the default to be true when unset, true (explicity set), or false.

Reply:
That does work but I always worry about issues with defaults actually being initialized and not accidentally cleared to false before the unpack. I prefer to be safer with the `*bool`.